internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -485,261 +475,6 @@ func (md *machineDeployment) updateUsingBlueGreenStrategy(ctx context.Context, u
 	return nil
 }
 
-func (md *machineDeployment) updateUsingImmediateStrategy(parentCtx context.Context, updateEntries []*machineUpdateEntry) error {
-	parentCtx, span := tracing.GetTracer().Start(parentCtx, "immediate")
-	defer span.End()
-
-	sl := statuslogger.Create(parentCtx, len(updateEntries), true)
-	defer sl.Destroy(false)
-
-	updatesPool := pool.New().WithErrors().WithContext(parentCtx)
-	if md.maxConcurrent > 0 {
-		updatesPool = updatesPool.WithMaxGoroutines(md.maxConcurrent)
-	}
-
-	for i, e := range updateEntries {
-		e := e
-		eCtx := statuslogger.NewContext(parentCtx, sl.Line(i))
-		fmtID := e.leasableMachine.FormattedMachineId()
-		statusRunning := func() {
-			statuslogger.LogfStatus(eCtx,
-				statuslogger.StatusRunning,
-				"Updating %s",
-				md.colorize.Bold(fmtID),
-			)
-		}
-		statusFailure := func(err error) {
-			if errors.Is(err, context.Canceled) {
-				statuslogger.LogfStatus(eCtx,
-					statuslogger.StatusFailure,
-					"Machine %s update %s",
-					md.colorize.Bold(fmtID),
-					md.colorize.Red("canceled while it was in progress"),
-				)
-			} else {
-				statuslogger.LogfStatus(eCtx,
-					statuslogger.StatusFailure,
-					"Machine %s update %s: %s",
-					md.colorize.Bold(fmtID),
-					md.colorize.Red("failed"),
-					err.Error(),
-				)
-			}
-		}
-		statusSuccess := func() {
-			statuslogger.LogfStatus(eCtx,
-				statuslogger.StatusSuccess,
-				"Machine %s update %s",
-				md.colorize.Bold(fmtID),
-				md.colorize.Green("succeeded"),
-			)
-		}
-
-		updatesPool.Go(func(_ context.Context) error {
-			statusRunning()
-			if err := md.updateMachine(eCtx, e); err != nil {
-				tracing.RecordError(span, err, "failed to update machine")
-				statusFailure(err)
-				return err
-			}
-			statusSuccess()
-			return nil
-		})
-	}
-
-	return updatesPool.Wait()
-}
-
-func (md *machineDeployment) updateUsingRollingStrategy(parentCtx context.Context, updateEntries []*machineUpdateEntry) error {
-	parentCtx, span := tracing.GetTracer().Start(parentCtx, "rolling", trace.WithAttributes(attribute.String("strategy", md.strategy)))
-	defer span.End()
-
-	sl := statuslogger.Create(parentCtx, len(updateEntries), true)
-	defer sl.Destroy(false)
-
-	// Rolling strategy
-	slices.SortFunc(updateEntries, func(a, b *machineUpdateEntry) int {
-		return cmp.Compare(a.leasableMachine.Machine().ID, b.leasableMachine.Machine().ID)
-	})
-
-	// Group updates by process group
-	entriesByGroup := lo.GroupBy(updateEntries, func(e *machineUpdateEntry) string {
-		return e.launchInput.Config.ProcessGroup()
-	})
```

Comment:
the whole point here is to do controlled concurrent updates per process group. I can't find that in new code

Reply:
yea that's a good catch!